<launch>

    <node pkg="coax_server"
          name="coax"
          type="coax_server"
	  output="screen"
          args="/dev/ttyS0:1">
      <remap from="/coax_server/state" to="/state"/>

      <remap from="/coax_server/control" to="/control"/>
      <remap from="/coax_server/rawcontrol" to="/rawcontrol"/>

      <remap from="/coax_server/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/coax_server/configure_comm" to="/configure_comm"/>
      <remap from="/coax_server/set_timeout" to="/set_timeout"/>
    </node>

    <node pkg="coax_gumstix_control"
          name="coax_gumstix_control"
          type="coax_gumstix_control"
	  output="screen">
      <remap from="/coax_gumstix_control/odom" to="/odom_smooth"/>
      <remap from="/coax_gumstix_control/state" to="/state"/>
      <remap from="/coax_gumstix_control/fmdes" to="/fmdes"/>
      <remap from="/coax_gumstix_control/nav_mode" to="/matlab_nav_mode"/>

      <remap from="/coax_gumstix_control/rawcontrol" to="/rawcontrol"/>
      <remap from="/coax_gumstix_control/info" to="/coax_info"/>
      <remap from="/coax_gumstix_control/imu" to="/coax_imu"/>
      <remap from="/coax_gumstix_control/control_mode" to="/control_mode"/>

      <remap from="/coax_gumstix_control/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/coax_gumstix_control/configure_comm" to="/configure_comm"/>
      <remap from="/coax_gumstix_control/set_timeout" to="/set_timeout"/>
      <remap from="/coax_gumstix_control/set_control_mode" to="/set_control_mode"/>

      <param name="frequency" value="100"/>
      <param name="CoaX" value="56"/>

      <rosparam file="$(find coax_gumstix_control)/config/coax_parameters.yaml"/>
      <rosparam file="$(find coax_gumstix_control)/config/coax_control_params.yaml"/>
    </node>

</launch>
